server/auth_test.go

```diff
@@ -229,3 +219,142 @@ func TestSwagger(t *testing.T) {
+	none := "none-" + ts
+	viewer := "viewer-" + ts
+	editor := "editor-" + ts
+	admin := "admin-" + ts
+
+	noneID := createUser(t, none)
+	defer deleteUser(t, noneID)
+
+	viewerID := createUserWithRole(t, viewer, "Viewer")
+	defer deleteUser(t, viewerID)
+
+	editorID := createUserWithRole(t, editor, "Editor")
+	defer deleteUser(t, editorID)
+
+	adminID := createUserWithRole(t, admin, "Admin")
+	defer deleteUser(t, adminID)
+
+	tests := []struct {
+		name       string
+		url        string
+		method     string
+		login      string
+		statusCode int
+	}{
+		{name: "settings-default", url: "/v1/Settings/Get", method: http.MethodPost, login: none, statusCode: http.StatusUnauthorized},
+		{name: "settings-viewer", url: "/v1/Settings/Get", method: http.MethodPost, login: viewer, statusCode: http.StatusUnauthorized},
+		{name: "settings-editor", url: "/v1/Settings/Get", method: http.MethodPost, login: editor, statusCode: http.StatusUnauthorized},
+		{name: "settings-admin", url: "/v1/Settings/Get", method: http.MethodPost, login: admin, statusCode: http.StatusOK},
+		{name: "alerts-default", url: "/alertmanager/api/v2/alerts", method: http.MethodGet, login: none, statusCode: http.StatusUnauthorized},
+		{name: "alerts-viewer", url: "/alertmanager/api/v2/alerts", method: http.MethodGet, login: viewer, statusCode: http.StatusUnauthorized},
+		{name: "alerts-editor", url: "/alertmanager/api/v2/alerts", method: http.MethodGet, login: editor, statusCode: http.StatusUnauthorized},
+		{name: "alerts-admin", url: "/alertmanager/api/v2/alerts", method: http.MethodGet, login: admin, statusCode: http.StatusOK},
+	}
+
+	for _, test := range tests {
+		test := test
+		t.Run(test.name, func(t *testing.T) {
+			// make a BaseURL without authentication
+			u, err := url.Parse(pmmapitests.BaseURL.String())
+			require.NoError(t, err)
+			u.User = url.UserPassword(test.login, test.login)
+			u.Path = test.url
+
+			req, err := http.NewRequest(test.method, u.String(), nil)
+			require.NoError(t, err)
+
+			resp, err := http.DefaultClient.Do(req)
+			require.NoError(t, err)
+			defer resp.Body.Close() //nolint:errcheck
+
+			assert.Equal(t, test.statusCode, resp.StatusCode)
+		})
+	}
+}
+
+func doRequest(t testing.TB, client *http.Client, req *http.Request) (*http.Response, []byte) {
+	resp, err := client.Do(req)
+	require.NoError(t, err)
+
+	defer resp.Body.Close() //nolint:errcheck
+
+	b, err := ioutil.ReadAll(resp.Body)
+	require.NoError(t, err)
+
+	return resp, b
+}
+
+func createUserWithRole(t *testing.T, login, role string) int {
+	userID := createUser(t, login)
+	setRole(t, userID, role)
+
+	return userID
+}
+
+func deleteUser(t *testing.T, userID int) {
+	u, err := url.Parse(pmmapitests.BaseURL.String())
+	require.NoError(t, err)
+	u.Path = "/graph/api/admin/users/" + strconv.Itoa(userID)
+
+	req, err := http.NewRequest(http.MethodDelete, u.String(), nil)
+	require.NoError(t, err)
+
+	req.Header.Set("Content-Type", "application/json; charset=utf-8")
```

Comment:
Looks a bit weird – do we really have to set Content-Type for bodiless DELETE request?

Reply:
Removed